src/components/Registration/Registration.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import './Registration.css';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { withAlert } from 'react-alert';
+import { compose } from 'redux';
+
+export class Registration extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      registerError: false,
+    };
+  }
+
+  handleInputChange = (e) => {
+    const { name, value } = e.target;
+    this.setState(() => ({ [name]: value }));
+  };
+
+  isValidEmail = (email) => {
+    const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
+      doesEmailRegexMatchEmail = emailRegex.test(email);
+    return doesEmailRegexMatchEmail;
+  };
+
+  isValidPassword = (password) => {
+    const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,32}$/,
+      doesPasswordRegexMatchPassword = passwordRegex.test(password);
+    return doesPasswordRegexMatchPassword;
+  };
+
+  registerUser = async () => {
+    const {
+        firstName, lastName, email, password,
+      } = this.state,
+      requestBody = {
+        firstName,
+        lastName,
+        email,
+        password,
+      };
+
+    await axios.post('http://localhost:5005/api/v1.0/users', requestBody);
```

Comment:
We should do all API calls inside the api.js file and call methods from there. That way if we ever need to make the same API call, we can just call the function that is made for us.